compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -159,7 +159,7 @@ struct MacroRulesMacroExpander {
     name: Ident,
     span: Span,
     transparency: Transparency,
-    lhses: Vec<mbe::TokenTree>,
+    lhses: Vec<Result<Vec<MatcherLoc>, ()>>,
```

Comment:
```suggestion
    lhses: Vec<Vec<MatcherLoc>>,
```
`Err`s here are treated as `span_bug`s by the logic below, so there should be no need to keep `Result`s here.

Reply:
Good point, I will fix this.

Reply:
Oh, I remember why I did it this way. I was getting ICEs with malformed matchers like `x => (3)`. But I can handle that by only converting to `MatcherLoc` form in `compile_declarative_macro` if `valid` is true, and using an empty `Vec` instead.